Detectors/EMCAL/simulation/src/LZEROElectronics.cxx

```diff
@@ -0,0 +1,241 @@
+      auto whichFastOrTRU2 = std::get<1>(mTriggerMap->getTRUFromAbsFastORIndex(fastor));
+
+      if (digitsList.size() == 0) {
+        continue;
+      }
+      digitsList.sort();
+
+      int digIndex = 0;
+      Digit summedDigit;
+      bool first = true;
+      for (auto& ld : digitsList) {
+        if (first) {
+          summedDigit = ld;
+          first = false;
+        } else {
+          // summedDigit += ld;
+
+          // safety device in case same fastOr
+          // but different towers, i.e. remember
+          // that the += operator fails WITHOUT
+          // feedback if that were to happen
+          Digit digitToSum(summedDigit.getTower(), ld.getAmplitude(), summedDigit.getTimeStamp());
+          summedDigit += digitToSum;
+        }
+      }
+
+      sizemDigitMap = (*digitsTimeBin.mDigitMap).size();
+      if (mSimulateNoiseDigits) {
+        addNoiseDigits(summedDigit);
+      }
+
+      if (summedDigit.getAmplitude() < 0) {
+        continue;
+      }
+
+      auto whichTRU = std::get<0>(mTriggerMap->getTRUFromAbsFastORIndex(fastor));
+      auto whichFastOrTRU = std::get<1>(mTriggerMap->getTRUFromAbsFastORIndex(fastor));
+
+      auto whichFastOr = std::get<1>(mTriggerMap->convertFastORIndexTRUtoSTU(whichTRU, whichFastOrTRU));
+      auto& patchTRU = patchesFromAllTRUs[whichTRU];
+      auto& fastOrPatchTRU = patchTRU.mFastOrs[whichFastOr];
+      fastOrPatchTRU.updateADC(summedDigit.getAmplitudeADC());
+
+      digIndex++;
+      counterhelp++;
+    }
+
+    // Evaluate -> peak finder (ALL Patches in ALL TRUs)
+    // in case peak found:
+    // - Create trigger input (IR of that timebin - delay [typically 8 or 9 samples - rollback (0)])
+    // - Create L1 timesums (trivial - last time integral) -> Collect from all fastOrs in ALL TRUs
+
+    // Trigger Inputs needs to have the correction of the delay
+    // Propagating for now all the interaction record
+    // The typical delay is 8 to 9 BCs and the rollback as well
+    // The rollback is taken from EMCALReconstruction/RecoParam.h
+    // The delay is due to the interactions between EMCAL and CTP
+    // It accounts for the difference in times between L0a, L0b, and then there will be a L1 and L1b delay
+    // There is 1BC uncertainty on the trigger readout due to steps in the interaction between CTP and detector simulations
+    bool foundPeak = false;
+    for (auto& patches : patchesFromAllTRUs) {
+      updatePatchesADC(patches);
+      bool foundPeakCurrentTRU = peakFinderOnAllPatches(patches);
+      auto firedPatches = getFiredPatches(patches);
+      if (foundPeakCurrentTRU)
+        foundPeak = true;
+    }
+
+    if (foundPeak == true)
+      LOG(debug) << "DIG TRU fill in LZEROElectronics: foundPeak = " << foundPeak;
+    EMCALTriggerInputs TriggerInputsForL1;
+    if (foundPeak) {
+      TriggerInputsForL1.mInterRecord = record;
+      int whichTRU = 0;
+      for (auto& patches : patchesFromAllTRUs) {
+        if (whichTRU < 52) {
```

Comment:
We have at max. 52  TRUs, condition should be always true. In case of invalid indices the TriggerMapping throws exceptions.

Reply:
I have removed the `if`.